dace/sdfg/state.py

```diff
@@ -674,9 +675,16 @@ def arglist(self, defined_syms=None, shared_transients=None) -> Dict[str, dt.Dat
 
         # Add scalar arguments from free symbols
         defined_syms = defined_syms or self.defined_symbols()
+        # Fix for dynamic map inputs appearing in free symbols
+        from dace.sdfg.utils import dynamic_map_inputs
+        dynamic_inputs = set()
+        for node in self.nodes():
+            if isinstance(node, nd.MapEntry):
+                dynamic_inputs.update([e.dst_conn for e in dynamic_map_inputs(self.graph, node)])
+        free_symbols = self.free_symbols - dynamic_inputs
```

Comment:
I don't remember adding something in here but without these changes I get an undefined symbol error. Same goes for the changes below. I didn't test of all of them or only one is required.